### PR TITLE
feat: add password hashing and decryption to web

### LINKED ISSUE #1496: Make lib file that contains the hashing/encryption algorithm for the email/password and wallet decryption
### Overview

We need to encrypt the email and password and send that off to the server to properly compare against the DB.

### References and additional details

https://www.notion.so/shapeshift/Legacy-Native-Wallet-Migration-4164af0d0ad641faaee4c32a9aad2d80#3588648cfcd340f1bd39f90b7f64b331

### Acceptance Criteria

* These functions are tested
* The encryption matches byte for byte what mobile does.
* The wallet can successfully be decrypted.

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by theoboldfrazier**
## Description

This allows for the decryption of a wallet given a cipher string, email and password.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #1496 

## Risk

No risk, code not used, yet.

## Testing

Can be tested once there is a backend to respond with an encryptedWallet.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -131,6 +131,10 @@
     ]
   },
   "jest": {
+    "transform": {
+      "^.+\\.ts?$": "ts-jest"
+    },
+    "verbose": true,
```
Comment (cjthompson) [inline package.json:137]: Is this change necessary for this PR?

File: package.json [hunk 2]
```diff
@@ -197,6 +202,7 @@
     "eslint-plugin-prettier": "^4.0.0",
     "eslint-plugin-simple-import-sort": "^7.0.0",
     "fast-json-stable-stringify": "^2.1.0",
+    "jest": "^27.5.1",
```
Comment (cjthompson) [inline package.json:205]: This should already be included as part of react-scripts.

File: package.json [hunk 3]
```diff
@@ -208,6 +214,7 @@
     "reselect-tools": "^0.0.7",
     "ssri": "^8.0.1",
     "storybook-dark-mode": "^1.0.9",
+    "ts-jest": "^27.1.4",
```
Comment (cjthompson) [inline package.json:217]: Included with "react-scripts"

**Review by cjthompson — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (theoboldfrazier) [reply package.json]: removed.

File: package.json (hunk 3 above)
Comment (theoboldfrazier) [reply package.json]: You won't be able to use it with the jest transform config unless it's specified as a dev-dep directly. But I've removed that from this PR since it's unrelated and removed this.

File: package.json (hunk 1 above)
Comment (theoboldfrazier) [reply package.json]: Removed.

**Review by 0xdef1cafe — APPROVED** [review]
